src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -566,10 +605,77 @@ OverlayImpl::onPeerDeactivate (Peer::id_t id,
     m_publicKeyMap.erase(publicKey);
 }
 
-/** The number of active peers on the network
-    Active peers are only those peers that have completed the handshake
-    and are running the Ripple protocol.
-*/
+void
+OverlayImpl::onManifests (Job&,
+    std::shared_ptr<protocol::TMManifests> const& inbox,
+        std::shared_ptr<PeerImp> const& from)
+{
+    auto const n = inbox->list_size();
+    auto const& journal = from->pjournal();
+
+    if (journal.debug) journal.debug
+        << "TMManifest, " << n << (n == 1 ? " item" : " items");
+
+    protocol::TMManifests outbox;
+
+    for (std::size_t i = 0; i < n; ++i)
+    {
+        auto const& manifest = inbox->list().Get(i);
+        auto const& s = manifest.stobject();
+        STObject st(sfGeneric);
+        try
+        {
+            SerialIter sit(s.data(), s.size());
+            st.set(sit);
+        }
+        catch(...)
+        {
+            if (journal.info) journal.info
+                << "Bad manifest #" << i + 1;
+            continue;
+        }
+        auto const pk = get<AnyPublicKey>(st, sfPublicKey);
+        auto const seq = get(st, sfSequence);
+        if (! pk ||
+            ! seq ||
+            ! get(st, sfSignature) ||
+            ! get(st, sfSigningPubKey))           
+        {
+            if (journal.info) journal.info
+                << "Malformed manifest #" << i + 1;
+            continue;
+        }
+        // `continue` if pk is untrusted
+        if (! manifestCache_.maybe_accept (*pk, *seq, s, st, journal))
+        {
+            if (journal.warning) journal.warning
+                << "Ignored manifest seq #" << *seq << " for " << *pk;
+            continue;
+        }
+        if (journal.warning) journal.warning
```

Comment:
Per recent discussion, I don't think the ignore/accept/forward messages should be logged at warning level. Maybe one step down.